test/Common/MoistThermodynamics/runtests.jl

```diff
@@ -121,218 +121,193 @@ using LinearAlgebra
   # Exner function. FIXME: add correctness tests
   p=FT(1.e5); q_tot=FT(0.23)
   @test exner_given_pressure(p, PhasePartition(q_tot)) isa typeof(p)
-end
+# end
+
 
+# @testset "moist thermodynamics - constructor consistency" begin
 
-@testset "moist thermodynamics - constructor consistency" begin
   # Make sure `ThermodynamicState` arguments are returned unchanged
 
-  # FIXME/TODO: Increasing `n` to 40 results in broken tests
-  # due to failure in saturation adjustment. This needs
-  # to be improved/made more robust.
-  n = 35
   FT = Float64
-  θ_liq_ice = collect(range(250,400,length=n))
-  T = collect(range(250,400,length=n))
-  ρ = collect(range(0.5,2.0,length=n))
-  q_tot_dry = zeros(n)
-  q_tot_moist = collect(range(0.0,0.2,length=n))
-  q_liq = collect(range(0.0,0.1,length=n))
-  q_ice = collect(range(0.0,0.05,length=n))
-
-  e_int = collect(range(10_000.0,200_000.0,length=n))
-  p = collect(range(10_000.0,200_000.0,length=n))
-  q_pt_dry = PhasePartition.(q_tot_dry, q_liq .* 0, q_ice .* 0)
-  q_pt_moist = PhasePartition.(q_tot_moist, q_liq, q_ice)
-
-  # air_temperature_from_liquid_ice_pottemp_given_pressure-liquid_ice_pottemp inverse
-  θ_liq_ice = liquid_ice_pottemp_given_pressure.(T, p, q_pt_dry)
-  @test all(air_temperature_from_liquid_ice_pottemp_given_pressure.(θ_liq_ice, p, q_pt_dry) .≈ T)
-
-  # liquid_ice_pottemp-air_temperature_from_liquid_ice_pottemp_given_pressure inverse
-  T = air_temperature_from_liquid_ice_pottemp_given_pressure.(θ_liq_ice, p, q_pt_dry)
-  @test all(liquid_ice_pottemp_given_pressure.(T, p, q_pt_dry) .≈ θ_liq_ice)
-
-  # Accurate but expensive `LiquidIcePotTempSHumNonEquil` constructor (Non-linear temperature from θ_liq_ice)
-  T_non_linear = air_temperature_from_liquid_ice_pottemp_non_linear.(θ_liq_ice, ρ, q_pt_moist)
-  e_int_ = internal_energy.(T_non_linear, q_pt_moist)
-  ts = PhaseNonEquil.(e_int_, ρ, q_pt_moist)
-  @test all(T_non_linear .≈ air_temperature.(ts))
-  @test all(θ_liq_ice .≈ liquid_ice_pottemp.(ts))
+  e_int, ρ, q_tot, q_pt, T, p, θ_liq_ice = MT.tested_convergence_range(FT, 50)
 
   # PhaseDry
   ts = PhaseDry.(e_int, ρ)
   @test all(internal_energy.(ts) .≈ e_int)
   @test all(air_density.(ts) .≈ ρ)
 
-  # PhaseEquil: Dry case
-  ts = PhaseEquil.(e_int, ρ, q_tot_dry)
+  # PhaseEquil
+  ts = PhaseEquil.(e_int, ρ, q_tot, FT(1e-2), 90, Ref(MT.saturation_adjustment_SecantMethod))
   @test all(internal_energy.(ts) .≈ e_int)
```

Comment:
If the approximately equal sign here tests to machine precision, this cannot work when nonlinear saturation adjustment is active. You can expect a similar relative error on energy and other thermodynamic quantities as on temperature. So, this can only be accurate to within `e_int * (tol/T)`, and it should only be tested to this accuracy. Similarly for other thermodynamic quantities.

Reply:
There is no error for `e_int`, since this is an input and should be exactly equal when retrieved :)

Reply:
Temperature is really the only variable that we need to check for a physical tolerance, since this is determined via saturation adjustment. We can test additional variables (e.g., pressure), but it will follow from temperature.

Reply:
Right. But isn't l. 170 testing the internal energy computed from the derived temperature, which is computed by saturation adjustment and hence potentially inaccurate?